Src/Campus.Services/Core/EventService.cs

```diff
@@ -0,0 +1,147 @@
+                    eventViewDto.End = calendarEvent.EndDate?.ToString("yyyy-MM-ddTHH:mm");
+                }
+
+                eventViews.Add(eventViewDto);
+            }
+
+            return eventViews;
+        }
+
+        public async Task<int> AddEvent(string userId, EventAddDto eventDto)
+        {
+            Classroom defaultClassroom = await GetDefaultClassroomByUserId(userId);
+
+            var newEvent = new Event
+            {
+                Title = eventDto.Title,
+                Description = eventDto.Description,
+                StartDate = eventDto.Start,
+                EndDate = eventDto.End,
+                ActualLocation = eventDto.Location,
+                AllDay = eventDto.AllDay
+            };
+
+            defaultClassroom.Events.Add(newEvent);
+
+            await _context.SaveChangesAsync();
+
+            return newEvent.Id;
+        }
+
+        public async Task EditEventById(string userId, EventEditDto eventDto)
+        {
+            Classroom defaultClassroom = await GetDefaultClassroomByUserId(userId);
+
+            Event eventToEdit = defaultClassroom.Events
+                .FirstOrDefault(e => e.Id.ToString() == eventDto.Id);
+
+            if (eventToEdit == null)
+                throw new ApplicationException($"Event with {eventDto.Id} id not found for given user");
+
+            eventToEdit.Title = eventDto.Title;
+            eventToEdit.Description = eventDto.Description;
+            eventToEdit.StartDate = eventDto.Start;
+            eventToEdit.EndDate = eventDto.End;
+            eventToEdit.ActualLocation = eventDto.Location;
+            eventToEdit.AllDay = eventDto.AllDay;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task DeleteEventById(string userId, string eventId)
+        {
+            Classroom defaultClassroom = await GetDefaultClassroomByUserId(userId);
+
+            Event eventToDelete = defaultClassroom.Events
+                .FirstOrDefault(e => e.Id.ToString() == eventId);
+
+            if (eventToDelete == null)
+                throw new ApplicationException($"Event with {eventId} id not found for given user");
+
+            _context.Events.Remove(eventToDelete);
+
+            await _context.SaveChangesAsync();
+        }
+
+        private async Task<Classroom> GetDefaultClassroomByUserId(string userId)
+        {
+            var user = await _userManager.FindByIdAsync(userId);
+
+            if (user == null)
+                throw new ApplicationException("User with this ID doesn't exist");
+
+            await _context.Entry(user)
+                .Collection(u => u.Participation)
+                .LoadAsync();
+
+            Participant defaultParticipation = user.Participation.FirstOrDefault();
+
+            if (defaultParticipation == null)
+                throw new ApplicationException("Given user doesn't participate in any classroom");
+
+            await _context.Entry(defaultParticipation)
```

Comment:
`Recommendation`: Merge async loads with [Task.WhenAll()](https://jeremylindsayni.wordpress.com/2019/03/11/using-async-await-and-task-whenall-to-improve-the-overall-speed-of-your-c-code/)